include/simdjson/inline/document_stream.h

```diff
@@ -143,10 +143,14 @@ inline void document_stream::start() noexcept {
   next();
 }
 
+really_inline size_t document_stream::iterator::current_index() noexcept {
+  return stream.doc_index;
+}
 inline void document_stream::next() noexcept {
   if (error) { return; }
 
   // Load the next document from the batch
+  doc_index = batch_start + parser.implementation->structural_indexes[parser.implementation->next_structural_index];
```

Comment:
Note that this is set here, but if the parsing fails, for whatever reason, it might be invalid.

Reply:
If the parsing fails, it'll still represent the beginning of the document that failed, right?

Reply:
It should, yes. That's why I set it first thing, before calling stage 2 parsing.